foundation-ui/factories/ServiceFactory.js

```diff
@@ -0,0 +1,23 @@
+/*
+  Service Factory
+ */
+function ServiceFactory(hooks) {
+  return {
+    on(mountId /* , callback */) {
```

Comment:
If this is going to be used as the core for all Services, it might make more sense to rename the first argument to something other than `mountId`. Also I think we should explicitly define the arguments it can take so it's easier for people to trace down the api. Let's add in `, callback, priority)` explicitly.